pkg/reconciler/helper.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reconciler
+
+import (
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type Callback func(interface{})
+
+// TODO(mattmoor): Move this into knative/pkg/controller
+func EnsureTypeMeta(f Callback, gvk schema.GroupVersionKind) Callback {
```

Comment:
Can we just place the help func into knative/pkg here from the very beginning?
to avoid introducing the testing dependency.

Reply:
I was trying to keep this change relatively small for a cherry-pick, but that's on my TODO list for today.

Reply:
I have no objection to this.
But I'll leave it for @jonjohnsonjr to give LGTM